src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -168,22 +168,33 @@ public static double BitIncrement(double x)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static double CopySign(double x, double y)
         {
-            const long signMask = 1L << 63;
+            const double signMask = -0.0;
 
-            if (Sse.X64.IsSupported)
+            if (Sse.IsSupported || AdvSimd.IsSupported)
             {
                 // Create vectors of the elements, and make them float to allow using SSE rather than SSE2 instructions
                 var xvec = Vector128.CreateScalarUnsafe(x).AsSingle();
                 var yvec = Vector128.CreateScalarUnsafe(y).AsSingle();
 
-                // Remove the sign from x, and remove everything but the sign from y
-                // Creating from a 'const long' is better than from the correct 'const float/double'
                 var mask = Vector128.CreateScalarUnsafe(signMask).AsSingle();
-                yvec = Sse.And(yvec, mask);
-                xvec = Sse.AndNot(mask, xvec);
 
-                // Simply OR them to get the correct sign
-                return Sse.Or(xvec, yvec).AsDouble().ToScalar();
+                if (Sse41.IsSupported)
```

Comment:
Some of these are common patterns.. This, for example, is really just cross-platform `ConditionalSelect(condition, left, right)`

It might be a good idea to create some internal helpers that we can reuse more generally. It would help make this code more readable and would avoid duplicating the logic every time we need it.

I'd suggest just making it an `internal static` method on `Vector128` (not `Vector128<T>`)

Reply:
so, we would have:
```
Vector128<float> ConditionalSelect(Vector128<float> condition, Vector128<float> left, Vector128<float> right)
{
    if (Sse.IsSupported)
    {
        if (Sse41.IsSupported)
        {
            return Sse41.BlendVariable(...);
        }

        return And, AndNot, Or pattern;
    }
    else if (AAdvSimd.IsSupported)
    {
        return AdvSimd.BitwiseSelect(...);
    }
    else
    {
        ThrowHelpers.PNSE();
        return default;
    }
}
```

This code would then become:
```
if (Sse.IsSupported || AdvSimd.IsSupported)
{
    var mask = ...;
    var xvec = ...;
    var yvec = ...;
    return Vector128.ConditionalSelect(mask, xvec, yvec);
}
```